### PR TITLE
add redis command cheat sheet (continues abandoned #1210)

### PR DESCRIPTION
**Pull request by seisfeld**
Filled out PR template as per @MrChrisW's request.
Mirrored the sections given by [Filter by group](http://redis.io/commands#) and removed the '+' between command words as suggested by @zachthompson.
Swapped the template to 'reference' to make it look better.
I also added a few missing bits.

**What does your Instant Answer do?**
Shows Redis command reference.

**What is the data source for your Instant Answer? (Provide a link if possible)**
http://redis.io/commands

**Why did you choose this data source?**
It was given in abandoned PR #1210.

**Are there any other alternative (better) data sources?**
No idea. Probably not as this is the vendors site.

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
DB admins, programmers...

**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**
Yes. https://duck.co/ideas/idea/5987/redis-command-cheat-sheet

**Are you having any problems? Do you need our help with anything?**
No

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**
redis

-----
IA Page: https://duck.co/ia/view/redis_cheat_sheet

### CODE REVIEW

File: share/goodie/cheat_sheets/json/redis.json [hunk 1]
```diff
@@ -0,0 +1,617 @@
...
+    "Sets",
+    "Sorted Sets",
+    "Strings",
+    "Transactions"
+  ],
+  "template_type": "reference",
+  "sections": {
+    "Cluster": [
+      {
+        "key": "CLUSTER ADDSLOTS slot [slot ...]",
```
Comment (mintsoft) [inline share/goodie/cheat_sheets/json/redis.json]: Are these capitalised because it's a SQL-esque language or for emphasis?

**Comment by seisfeld** [comment]
@mintsoft I believe it's due to the fact that it's SQL style. It's the same at the source documents of course. I am no redis expert though.

File: share/goodie/cheat_sheets/json/redis.json [hunk 2]
```diff
@@ -0,0 +1,617 @@
+{
+  "id": "redis_cheat_sheet",
+  "name": "Redis commands",
+  "metadata": {
+    "sourceName": "redis.io/commands",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/redis.json]: We usually just go with the domain name, like `Redis`. I guess `Redis.io` could work.

**Comment by moollaza** [comment]
@seisfeld thanks a lot for continuing this PR. I'm a little concerned with the amount of content here though. Cheat Sheets shouldn't really be exhaustive unless it's a small amount of content. I'm not a Redis user so I'm not sure what the most important/common commands are, but do you think we can remove some unnecessary stuff from this, or would you argue that all of it is of equal importance?

**Comment by moollaza** [comment]
We could potentially break it up into multiple cheat sheets if you think there's a reasonable/appropriate way to split up the content

**Comment by seisfeld** [comment]
@moollaza I'm not a redis expert either. I just noticed you recently abandoned #1210 which looked more or less complete to me (Except for the suggestions made my @MrChrisW and @zachthompson). So i went ahead and worked on these. And here we are...

**Comment by zachthompson** [comment]
Nice @seisfeld thanks for resuscitating this cheat sheet.  With respect to breaking it up for digestibility, we could pull out cluster, server, etc. into an admin cheat sheet.  The data types and commands for working with them I imagine will be the most requested.  It will still be pretty big but it's not big like, say, putting an entire programming language in a single cheat sheet.

**Comment by seisfeld** [comment]
Let me know your decision and if this is doable I'll quickly implement it. I just want to avoid any double work. :smiley:

**Comment by zekiel** [comment]
here's your IA page (it will be filled in shortly w/ info from this PR) https://duck.co/ia/view/redis_cheat_sheet 

If you'd like to make it shorter, it would still be useful as a cheat sheet (e.g. essential commands) 
If you think it's all worth it, I would suggest an implementation like: https://duck.co/ia/view/parse_cron

**Comment by seisfeld** [comment]
@zekiel Sure i can make it shorter if someone tells me what is important and what is not. To me all commands make equally no sense. :wink: I am no redis user. I just wanted to complete this abandoned IA because it looked complete anyways except the few open tasks which i completed.

**Comment by moollaza** [comment]
@seisfeld sorry for the delay!

>I am no redis user. I just wanted to complete this abandoned IA because it looked complete anyways except the few open tasks which i completed. 

Understood, thanks for picking this up! I suggest you reach out to the Redis users in our DuckDuckHack Slack to have them review what the most important Redis commands are. Alternatively, I would suggest you have them provide you with an existing cheat sheet somewhere online (which is more succinct) and then you can copy that. Right now, it's simply too verbose. At 600 lines, this is basically a recreation of their docs!

**Comment by zachthompson** [comment]
Try removing these sections:

* Cluster
* Server
* Connection
* HyperLogLog
* Pub/Sub
* Scripting
* Server
* Transactions

**Comment by seisfeld** [comment]
Thanks @zachthompson. I'll remove these and update the PR.

**Comment by seisfeld** [comment]
@zachthompson @moollaza Here you go, i removed the sections. Also checked the travis output but honestly this is not conclusive to me. Please check. :smile:

**Comment by seisfeld** [comment]
Seems i could fix the travis issue myself. :+1:

**Comment by zachthompson** [comment]
Looks good to me.  Now it's only 6th biggest :D

File: share/goodie/cheat_sheets/json/redis.json [hunk 3]
```diff
@@ -0,0 +1,370 @@
+{
+  "id": "redis_cheat_sheet",
+  "name": "Redis commands",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/redis.json]: We should capitalize `Commands` so it's a proper title

File: share/goodie/cheat_sheets/json/redis.json (hunk 2 above)
Comment (moollaza) [inline share/goodie/cheat_sheets/json/redis.json]: This should be `Redis.io` or `Redis`

**Comment by moollaza** [comment]
@seisfeld this LGTM :+1:

Thanks a lot for contributing (we really appreciate it :smiley:). We invite you to stick around and comment/help on other PR's & Issues or even submit an Instant Answer!

**Comment by moollaza** [comment]
@seisfeld congrats! It's live: https://duckduckgo.com/?q=Redis+Cheat+Sheet&ia=cheatsheet

Thanks again for taking the time to contribute! We really appreciate it. Feel free to stick around and comment/help on other PR's, tackle open issues or even submit more Instant Answers!

**Comment by moollaza** [comment]
@itamarhaber apologies for the late reply, somehow your reply slipped through my GH notifications.

Also, happy new year!

>I think the current sheet should be added with HLL (http://redis.io/commands#hyperloglog) and Geo (http://redis.io/commands#geo) - both basically belong to the same class of the other data structures and won't increase the sheet's length significantly.

That sounds reasonable, thanks for the suggestion. I'll make a task for the community to add in content for HLL and Geo

Regarding the Redis Fathead, we're getting it back online -- it seems to be in working order, but the content may need to be updated. I'll let you know when it goes live!

**Pull request merged**